website/app/jobs/update_order_status.ts

```diff
@@ -61,10 +62,18 @@ export default class UpdateOrderStatus extends Job {
 
             const total = order.total
             const orderId = order.id
+            const user = await User.find(order.userId)
+            for (const product of products) {
+              const event = await Event.query().where('product_id', product.id).first()
+              if (event) {
+                const eventService = new EventService()
+                await eventService.register(user!, event)
```

Comment:
Given that just a few lines below you have a null-check for the `user` variable, I'd also add that here as a safety measure.